### ISSUE TITLE
Trip does not show although route was recorded

### ISSUE THREAD

**Issue by maximilianloy** [traccar/traccar#4301]
One of the trackers recorded a route last week (see screenshot) and Traccar properly reports the route, statistics, events,... but the trip report won't show anything. The API call returns an empty array (no error).
Trips generally work for us, also for those distances, but it is the first time that one does not simply show up. Also for this tracker we see older trips.

We are running Traccar 4.2 with no special configuration regarding trip calculation (using defaults)

<img width="1036" alt="Screenshot 2019-04-30 at 22 23 08" src="https://user-images.githubusercontent.com/6249264/56991364-c58b8680-6b97-11e9-9572-dadcaeeac60f.png">

**Comment by tananaev** [comment]
Trip calculation logic is pretty well documented here:

https://www.traccar.org/documentation/trips-stops/

**Comment by maximilianloy** [comment]
But that does not explain to me why in this case the trip was not detected. Duration was longer than 5 minutes (1,5h) and further than 500m (19km)

**Comment by tananaev** [comment]
Most common explanation is that you didn't have stop after the trip, which is one of the requirements.

**Comment by maximilianloy** [comment]
Yes it looks like the stop event was never triggered, although the tracker went offline:

<img width="333" alt="Screenshot 2019-05-01 at 00 05 31" src="https://user-images.githubusercontent.com/6249264/56996456-e1e1f000-6ba4-11e9-998e-f00fef08c755.png">

**Comment by maximilianloy** [comment]
Ok, I think I found the issue. Seems like the tracker was moving and went straight into an area with no GPS reception. It reported an invalid location from that point on.
We changed Traccar code so that also for invalid locations, in case the status (alarm) changes, it re-uses the last location but with the new location. Obviously we did not set the speed to 0, which resulted in Traccar thinking the device is still moving.
